### PR TITLE
[REF] mail: simplify discuss model insert code

### PR DESCRIPTION
**Pull request by alexkuhn**
1. simplify message reaction formatter (personas)

Data was formatted to have "partners" and "guests" entries, both
of which contribute to personas.

To avoid some post-processing of data in JS, it's best to format
data to immediately include type of persona.

2. include "guestAuthor" in "author" data of message

Discuss models in JS group partners and guests into a single model
Persona, to make feature works regardless on whether user is
authenticated or not.

This commit simplifies code by removing data `guestAuthor` in message
formatted data, and instead `author` contains author data in all cases,
whether the author is a partner or guest.

3. simplify insert (remove id, redundant with preinsert)

- Also rename `Follower.isActive` to `Follower.is_active`, for
even simpler `Follower.insert()`

### CODE REVIEW

File: addons/mail/static/src/core/common/messaging_service.js [hunk 1]
```diff
@@ -70,7 +70,6 @@ export class Messaging {
             this.store.guest = {
                 ...data.currentGuest,
                 type: "guest",
-                channelId: data.channels[0]?.id,
```
Comment (alexkuhn) [inline addons/mail/static/src/core/common/messaging_service.js:73]: Was deadcode

File: addons/mail/static/src/core/common/channel_member_model.js [hunk 2]
```diff
@@ -32,10 +32,8 @@ export class ChannelMember extends Record {
         this.id = data.id;
         if ("persona" in data) {
             this.persona = {
-                ...(data.persona.partner ?? data.persona.guest),
-                type: data.persona.guest ? "guest" : "partner",
-                country: data.persona.partner?.country,
-                channelId: data.persona.guest ? data.channel.id : null,
+                ...data.persona,
+                country: data.persona?.country,
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/channel_member_model.js:36]: Isn't it a dummy assignation? (the 2nd one) As persona is already spread and the key has the same name.

File: addons/mail/static/src/core/common/message_reactions_model.js [hunk 3]
```diff
@@ -21,33 +21,9 @@ export class MessageReactions extends Record {
...
-            } else if (command !== "ADD") {
-                personasToUnlink.add(persona.localId);
-            }
-        }
         Object.assign(reaction, {
             count: data.count,
-            content: data.content,
-            message: data.message,
-            personas: reaction.personas.filter((p) => !personasToUnlink.has(p.localId)),
+            personas: data.personas,
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/message_reactions_model.js:26]: not yet possible to completely assign data?

**Review by seb-odoo — APPROVED** [review]
@robodoo rebase-ff delegate+

File: addons/mail/static/src/core/common/channel_member_model.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/common/channel_member_model.js:36]: I saw it and planned for follow-up PR, but indeed I can just change the code now

File: addons/mail/static/src/core/common/message_reactions_model.js (hunk 3 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/common/message_reactions_model.js:26]: Probably yes, will do the change now

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**